subprojects/test-runner/device-provider/src/main/kotlin/com/avito/android/runner/devices/internal/Wait.kt

```diff
@@ -13,9 +13,14 @@ internal suspend fun <T> waitForCondition(
     successMessage: String? = null,
     maxAttempts: Int = WAIT_FOR_COMMAND_MAX_ATTEMPTS,
     frequencySeconds: Long = WAIT_FOR_COMMAND_FREQUENCY_SECONDS,
+    timeoutSec: Long = WAIT_FOR_COMMAND_TIMEOUT,
     condition: suspend () -> Result<T>
 ): Result<T> {
-    for (attempt in 0 until maxAttempts) {
+    val timeoutMs = TimeUnit.SECONDS.toMillis(timeoutSec)
+    val startTime = System.currentTimeMillis()
+    var lastCycleTime = startTime
+    var attempt = 1
+    while (lastCycleTime - startTime < timeoutMs && attempt <= maxAttempts) {
         when (val result = condition()) {
             is Success -> {
                 logger.debug("${successMessage ?: "$conditionName succeed"} at attempt=$attempt")
```

Comment:
```suggestion
                val durationMs = System.currentTimeMillis() - startTime
                logger.debug("${successMessage ?: "$conditionName succeed"} in ${durationMs} at attempt=$attempt")
```
It may help in troubleshooting.